File: src/model/Card.java
package model;

import model.common.CardType;
import model.common.WalletObject;

public class Card extends WalletObject {
    private final CardType type;
    public Card(int id, CardType type) {
        super(id);
        this.type = type;
    }
    public CardType getType() {
        return type;
    }

    @Override
    public String toString() {
        return "Card{" +
                "type=" + type +
                '}';
    }
}

File: src/model/Card.dart
import './common/wallet_object.dart';
import './common/card_type.dart';

class Card extends WalletObject {
  final CardType type;

  Card(int id, this.type) : super(id);

  CardType getType() {
    return type;
  }

  @override
  String toString() {
    return 'Card{type=$type}';
  }
}
